src/c++11/span.h

```diff
@@ -47,19 +64,21 @@ class Span {
   // Gets the Tracer associated with this Span.
   Tracer tracer() const;
 
+  // Gets an immutable reference to this Span's context.
```

Comment:
Well the reference is immutable but it references a mutable context in the Span. I believe we've decided this is not OK, so some kind of copy-on-write will be needed to prevent this immutable reference from seeing mutations on the context it's derived from.